src/search.ts

```diff
@@ -0,0 +1,611 @@
+  channelName?: string
+}
+
+const getTypes = (activeFacets: Facets | undefined) => {
+  if (activeFacets && activeFacets.type) {
+    return activeFacets.type
+  } else {
+    return [SEARCH_FILTER_COMMENT, SEARCH_FILTER_POST, SEARCH_FILTER_PROFILE]
+  }
+}
+
+export const buildSearchQuery = ({
+  text,
+  from,
+  size,
+  sort,
+  activeFacets,
+  channelName
+}: SearchQueryParams): Record<string, any> => {
+  let builder = bodybuilder()
+
+  if (!isNil(from)) {
+    builder = builder.from(from)
+  }
+  if (!isNil(size)) {
+    builder = builder.size(size)
+  }
+  if (
+    sort &&
+    activeFacets &&
+    !(activeFacets.type ?? []).includes(LearningResourceType.ResourceFile)
+  ) {
+    const { field, option } = sort
+    const fieldPieces = field.split(".")
+
+    const sortQuery = {
+      order:  option,
+      nested: {
+        path: fieldPieces[0]
+      }
+    }
+
+    if (field === COURSENUM_SORT_FIELD) {
+      if ((activeFacets.department_name ?? []).length === 0) {
+        // @ts-ignore
+        sortQuery["nested"]["filter"] = {
+          term: {
+            "department_course_numbers.primary": true
+          }
+        }
+      } else {
+        const filterClause: any[] = []
+        addFacetClauseToArray(
+          filterClause,
+          "department_course_numbers.department",
+          activeFacets.department_name || [],
+          LearningResourceType.Course
+        )
+        // @ts-ignore
+        sortQuery["nested"]["filter"] = filterClause[0]
+      }
+    }
+
+    builder.sort(field, sortQuery)
+  }
+
+  const types = getTypes(activeFacets)
+  const searchText = normalizeDoubleQuotes(text)
+  return emptyOrNil(
+    intersection([...LR_TYPE_ALL, LearningResourceType.ResourceFile], types)
+  ) ?
+    buildChannelQuery(builder, searchText, types, channelName) :
+    buildLearnQuery(builder, searchText, types, activeFacets)
+}
+
+export const buildChannelQuery = (
+  builder: Bodybuilder,
+  text: string | null,
+  types: Array<string>,
+  channelName: string | undefined
+): Record<string, any> => {
+  for (const type of types) {
+    const textQuery = emptyOrNil(text) ?
+      {} :
+      {
+        should: [
+          {
+            multi_match: {
+              query:  text,
+              fields: searchFields(type as LearningResourceType)
+            }
+          }
+        ].filter(clause => clause !== null)
+      }
+
+    // If channelName is present add a filter for the type
+    const channelClauses = channelName ?
+      [
+        {
+          term: {
+            [channelField(type as LearningResourceType)]: channelName
+          }
+        }
+      ] :
+      []
+
+    builder = buildOrQuery(builder, type, textQuery, channelClauses)
+  }
+
+  if (!emptyOrNil(text)) {
+    builder = builder.rawOption(
+      "suggest",
+      // @ts-ignore
+      buildSuggestQuery(text, CHANNEL_SUGGEST_FIELDS) // @ts-ignore
+    )
+  }
+
+  return builder.build()
+}
+
+export const buildLearnQuery = (
```

Comment:
Here and in the other stuff that we definitely want to export, a JSDoc / TSDoc docstring would be really nice, even if it's very simple.